build-tools/packages/build-tools/src/genMonoRepoPackageJson/lib.ts

```diff
@@ -4,121 +4,82 @@
     const topLevelTotalCount = topLevelDevCount;
 
     const markNonDev = (name: string, topRef: string, item: any, refStack: any[]) => {
         totalDevCount--;
         delete item.dev;
         refStack.push(item);
-
-        if (item.dependencies !== undefined) {
+        if (item.dependencies) {
             // mark unhoisted dependencies recursively
-            // eslint-disable-next-line guard-for-in
             for (const dep in item.dependencies) {
                 markNonDev(dep, topRef, item.dependencies[dep], refStack);
             }
         }
-
         // Mark the hoisted dependencies
         for (const req in item.requires) {
-            if (!refStack.some((scope) => scope.dependencies?.[req] !== undefined)) {
+            if (!refStack.some(scope => scope.dependencies && scope.dependencies[req] !== undefined)) {
                 markTopLevelNonDev(req, name, topRef);
             }
         }
-
         refStack.pop();
-    };
+    }
 
     const markTopLevelNonDev = (dep: string, ref: string, topRef: string) => {
         const item = repoPackageLockJson.dependencies[dep];
-        if (item !== undefined) {
-            logger.errorLog(
-                `Missing ${dep} in lock file referenced by ${ref} from ${topRef} in ${monoRepo.kind.toLowerCase()}`,
-            );
+        if (!item) {
+            throw new Error(`Missing ${dep} in lock file referenced by ${ref} from ${topRef} in ${monoRepo.kind.toLowerCase()}`);
+        }
+        if (commonOptions.verbose) {
+            log?.verbose(`NonDev Ref: ${topRef}..${ref} => ${dep}`);
         }
 
-        logger.verbose(`NonDev Ref: ${topRef}..${ref} => ${dep}`);
-
-        if (item.dev !== undefined) {
+        if (item.dev) {
             topLevelDevCount--;
             markNonDev(dep, dep, item, []);
         }
-    };
+    }
 
     // Go thru the non-dev dependencies in the package.json file and recursively mark the dependency tree as non-dev
-    // eslint-disable-next-line guard-for-in
     for (const dep in repoPackageJson.dependencies) {
         markTopLevelNonDev(dep, "<root>", "<root>");
     }
 
-    logger.info(
-        `${monoRepo.kind}: ${format(totalDevCount)}/${format(totalCount)} locked devDependencies`,
-    );
-    logger.info(
-        `${monoRepo.kind}: ${format(topLevelDevCount)}/${format(
-            topLevelTotalCount,
-        )} top level locked devDependencies`,
-    );
-    return writeFileAsync(
-        path.join(monoRepo.repoPath, "repo-package-lock.json"),
-        JSON.stringify(repoPackageLockJson, undefined, 2),
-    );
+    log?.info(`${monoRepo.kind}: ${format(totalDevCount)}/${format(totalCount)} locked devDependencies`);
+    log?.info(`${monoRepo.kind}: ${format(topLevelDevCount)}/${format(topLevelTotalCount)} top level locked devDependencies`);
+    return writeFileAsync(path.join(monoRepo.repoPath, "repo-package-lock.json"), JSON.stringify(repoPackageLockJson, undefined, 2));
 }
 
 interface PackageJson {
```

Comment:
Is this defined anywhere else? Just wondering if this might be a redundant definition.

Reply:
There's an IPackage definition in fluid-build, but it's not exported atm so I'm just going to leave this.